coordinates.cpp

```diff
@@ -1 +1,47 @@
+#include "coordinates.h"
+#include <cmath> 
+#include <iostream>
+#include <vector> 
+#include <utility> 
 
+using namespace std;
+
+double Haversine(Coordinate a, Coordinate b) 
+    {   double rad = 6371; 
+        // find the distance between lat and long 
+        double latDistance = (b.first - a.first) * M_PI / 180.0; 
+        double longDistance = (b.second - a.second) * M_PI / 180.0;
+
+        // convert the distance to radians 
+        b.first = (b.first) * M_PI / 180.0; 
+        a.first = (a.first) * M_PI / 180.0; 
+
+        // find distance 
+        double powFormula; 
+        powFormula = pow(sin(latDistance / 2), 2) + pow(sin(longDistance /2), 2) * cos(b.first) * cos(a.first);
```

Comment:
you could use another line instead of have one long line. Like this -->
powFormula = pow(sin(latDistance / 2), 2) 
+ pow(sin(longDistance /2), 2) * cos(b.first) * cos(a.first);